cranelift/codegen/src/ir/trapcode.rs

```diff
@@ -14,6 +14,15 @@ pub enum TrapCode {
     /// The current stack space was exhausted.
     StackOverflow,
 
+    /// A load was performed to an unmapped address
+    OutOfBoundsLoad,
+
+    /// A store was performed to an unmapped address
+    OutOfBoundsStore,
+
+    /// An operation was performed with an invalid address
+    InvalidAddress,
+
```

Comment:
Is there a particular reason that we need to pull these specific errors out of the more general "heap out-of-bounds" error below? It seems to me that at least the first two (OOB load, OOB store) are canonical OOB cases, and "invalid address" would make some sense if we had a more general notion of address-space maps in Wasm, but I think the complexity and confusion presented in the API here is not really worth it. @abrown thoughts?

Reply:
I guess that makes sense: `InvalidAddress` is also really just an OOB. Can we get away with `StackOutOfBounds` here (to match the other `...OutOfBounds` below)?

Reply:
I didn't use `HeapOutOfBounds` because i thought it would be heap specific, but we can fold all other memory accesses / invalid address into that.

We can also specialize a `StackOutOfBounds` for `stack_loads`/`stores`.